adapter/internal/api/apis_impl.go

```diff
@@ -151,6 +174,31 @@ func extractAPIProject(payload []byte) (apiProject ProjectAPI, err error) {
 	return apiProject, nil
 }
 
+func verifyMandatoryFields(apiJSON config.APIJsonData) error {
+	var errMsg string = ""
+	if apiJSON.Data.APIName == "" {
+		errMsg = "API Name cannot be empty"
+	}
+
+	if apiJSON.Data.APIVersion == "" {
+		errMsg = "API Version cannot be empty"
+	}
+
+	if apiJSON.Data.APIContext == "" {
+		errMsg = "API Context cannot be empty"
+	}
+
+	if apiJSON.Data.EndpointConfig.ProductionEndpoints.Endpoint == "" &&
```

Comment:
Isn't API LifeCycleStatus and EndpointImplementationType is kinda mandatory according to the implementation?
Either we could check it here or else we could avoid being that field mandatory.

Reply:
API LifeCycleStatus and EndpointImplementationType are not  mandatory fields